### PR TITLE
dev/core#5172 SearchKit - Support relative dates in conditional rules

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Fixes https://lab.civicrm.org/dev/core/-/issues/5172 by allowing relative dates in SK conditionals.

### CODE REVIEW

File: ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php [hunk 1]
```diff
@@ -451,6 +451,16 @@ protected function getRuleCondition($clause, bool $isMulti = FALSE): array {
       }
       $fieldKey = $this->renameIfAggregate($fieldKey);
     }
+    // Handle relative dates
+    if (!empty($clause[2])) {
+      $field = $this->getField($fieldKey);
+      if ($field && $field['data_type'] === 'Date') {
+        $clause[2] = date('Y-m-d', strtotime($clause[2]));
```
Comment (jitendrapurohit) [inline ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php:458]: seems will still fail for relative dates?  eg

image

where $clause[2] is `this.day`. Might require modification to use CRM_Utils_Date helpers instead?

Comment (monishdeb) [reply ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php:458]: I think the `Date Range` option doesn't make sense with operators other then `Is between`. Suppose a end-user chose >= with previous calendar year then it translate the date range between 1-1-2023 to 31-12-2023 and thus any operator other than `Is between` isn't applicable. So I would say date range option should be enabled when `Is between` is selected. And when `Is between` is selected, like in Civi Reports date filter, there should be date range option and relative date options. Currently it shows From and To datepickers.

Comment (colemanw) [reply ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php:458]: This PR didn't get ambitious enough to support date ranges. Could do so as a followup but imo this pr is better in than out.

**Comment by Guydn** [comment]
The patch does not seem to work on 5.76.
Test: display a button (on a row) with a test on a date being LT or GT "now".

**Comment by colemanw** [comment]
@Guydn I have tested again and confirmed that this PR does work. Here is a screenshot of a button (on a row) set to GT "now":

image

**Comment by colemanw** [comment]
@Guydn I'm going to go ahead & merge this because it has a unit test and I've confirmed it in manual testing multiple times. I'm not sure if you had trouble applying the patch or what may have happened on your site, but I think having it merged will be helpful for you to get it onto your live site.

**Pull request merged**